ui/packages/consul-ui/app/components/data-source/index.js

```diff
@@ -169,6 +177,15 @@ export default class DataSource extends Component {
       }
     }
   }
+  @action
+  async invalidate() {
+    this.source.readyState = 2;
+    this.disconnect();
+    schedule('afterRender', () => {
+      // FIXME: Lazy data-sources
+      this.connect([]);
+    });
+  }
```

Comment:
Allows us to reload individual DataSources by pretending to remove them from the DOM and then pretending to re add them. We don't need/use this yet, but it will be handy later on. The FIXME here relates to not supporting `@loading="lazy"` DataSources yet. We can either do this here, or convert this to a TODO for merge. There's no rush to do this as we don't need it yet.